parsl/executors/taskvine/executor.py

```diff
@@ -330,11 +347,25 @@ def submit(self, func, resource_specification, *args, **kwargs):
             Keyword arguments to the Parsl app
         """
 
+        # a Parsl function must have a name
+        if func.__name__ is None:
+            raise ValueError('A Parsl function must have a name')
+
         logger.debug(f'Got resource specification: {resource_specification}')
 
         # Default execution mode of apps is regular
         exec_mode = resource_specification.get('exec_mode', self.function_exec_mode)
 
+        if exec_mode == 'serverless':
+            if func.__name__ not in self._map_func_names_to_func_details:
+                self._map_func_names_to_func_details[func.__name__] = {'func_obj': func}
+            else:
+                if id(func) != id(self._map_func_names_to_func_details[func.__name__]['func_obj']):
```

Comment:
I think it gives exactly the same comparison but I feel like if you want to compare two objects for actually being the same object, then:

`if func is self._map_func_names_to_func_details[func.__name__]['func_obj']`

might be a more fundamental test?

Although not an issue as the code is written here, it is stylistically resistant to id reuse over time, in case people later on get the idea that they can cache the result of `id()`.